python/clx/analytics/cybert.py

```diff
@@ -0,0 +1,227 @@
+        """
+        with open(self._vocabpath) as f:
+            for index, line in enumerate(f):
+                self._vocab_lookup[index] = line.split()[0]
+
+        with open(label_map_filepath) as f:
+            for index, line in enumerate(f):
+                self._label_map[index] = line.split()[0]
+
+        model_state_dict = torch.load(model_filepath)
+        self._model = BertForTokenClassification.from_pretrained(
+            pretrained_model,
+            state_dict=model_state_dict,
+            num_labels=len(self._label_map),
+        )
+        self._model.cuda()
+        self._model.eval()
+
+    def preprocess(self, raw_data_df, stride_len=116, max_seq_len=128):
+        """
+        Preprocess and tokenize data for cybert model inference.
+
+        :param raw_data_df: logs to be processed
+        :type model_filepath: cudf.Series
+        :param stride_len: Max stride length for processing, default is 116
+        :type stride_len: int
+        :param max_seq_len: Max sequence length for processing, default is 128
+        :type max_seq_len: int
+
+        Examples
+        --------
+        >>> import cudf
+        >>> from clx.analytics.cybert import Cybert
+        >>> cyparse = Cybert()
+        >>> cyparse.load_model('/path/to/model.pth', '/path/to/labels.txt')
+        >>> raw_df = cudf.Series(['Log event 1', 'Log event 2'])
+        >>> input_ids, attention_masks, meta_data = cyparse.preprocess(raw_df)
+        """
+        raw_data_df = raw_data_df.str.replace('"', "")
+        raw_data_df = raw_data_df.str.replace("\\r", " ")
+        raw_data_df = raw_data_df.str.replace("\\t", " ")
+        raw_data_df = raw_data_df.str.replace("=", "= ")
+        raw_data_df = raw_data_df.str.replace("\\n", " ")
+
+        self._byte_count = raw_data_df.str.byte_count()
+        self._max_num_chars = self._byte_count.sum()
+        self._max_rows_tensor = int((self._byte_count / 120).ceil().sum())
+
+        input_ids, att_mask, meta_data = raw_data_df.str.subword_tokenize(
+            self._hashpath,
+            128,
+            116,
+            max_num_strings=len(raw_data_df),
+            max_num_chars=self._max_num_chars,
+            max_rows_tensor=self._max_rows_tensor,
+            do_lower=False,
+            do_truncate=False,
+        )
+
+        num_rows = int(len(input_ids) / 128)
+        input_ids = from_dlpack(
+            (input_ids.reshape(num_rows, 128).astype(cupy.float)).toDlpack()
+        )
+        att_mask = from_dlpack(
+            (att_mask.reshape(num_rows, 128).astype(cupy.float)).toDlpack()
+        )
+        meta_data = meta_data.reshape(num_rows, 3)
+
+        return input_ids.type(torch.long), att_mask.type(torch.long), meta_data
+
+    def inference(self, raw_data_df):
+        """
+        Cybert inference and postprocessing on dataset
+
+        :param raw_data_df: logs to be processed
+        :type raw_data_df: cudf.Series
```

Comment:
If this is a series, maybe `raw_data_col` would be a better variable name. Open to suggestion